src/ui/header/Header.tsx

```diff
@@ -1,18 +1,21 @@
-import AppBar from '@mui/material/AppBar';
-import Box from '@mui/material/Box';
-import Toolbar from '@mui/material/Toolbar';
-import Typography from '@mui/material/Typography';
-import Button from '@mui/material/Button';
+import { AppBar, Box, Toolbar, Typography, Button } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
 
 import { useLocale } from 'hooks';
 import { HeaderProps } from './Header.types';
 import { LanguageSwitch } from './languageSwitch/LanguageSwitch';
 import { styles, StyledLink } from './Header.styles';
 import { AppRoute } from 'routing/AppRoute.enum';
 
-export const Header = ({ isLoggedIn }: HeaderProps) => {
+export const Header = ({ isLoggedIn, onLogoutClick }: HeaderProps) => {
   const { locale, setLocale, formatMessage } = useLocale();
 
+  const navigate = useNavigate();
+
+  const handleButtonClick = () => {
+    !isLoggedIn ? navigate(AppRoute.login) : onLogoutClick();
```

Comment:
Tak na przyszłość - teranry operator służy do zwracania wartości, a nie do warunkowego wywoływania funkcji. To tutaj jest po prostu nieczytelne. Wiem że `if` to więcej kodu, niemniej jest czytelny na pierwszy rzut oka